release.md

```diff
@@ -13,7 +13,7 @@ You should be part of the `scaffolding-codeowners` or `sigstore-oncall` groups,
 1. Click on the `+ Create new tag: v0.4.11 on publish` button that appears.
 1. For the release title, use the tag name: v0.4.11,
 1. Click the `Generate release notes` button
-1. Click the `Save Draft` button to trigger the `Release` action.
+1. Verify everything looks good and then clieck `Publish` button to trigger the `Release` action.
```

Comment:
we have goreleaser and that create the github release for us.
I think we can remove almost all this and rewrite something like

1. Create a tag with `git tag` and push the tag to upstream
2. This will trigger the release job
3. When the release job finish go to the github release and review the release and then if everything looks good click on `Publish`

Reply:
@cpanato does #488 look better? If that's the right direction, let's close this and use that instead?